### PR TITLE
feat: support wasm with no decl set

### LINKED ISSUE #860: [feat] generate web file in dart if dart_decl_output = None
**Is your feature request related to a problem? Please describe.**

generate web file in dart if dart_decl_output = None
```
let raw_opts = RawOpts {
        // Path of input Rust code
        rust_input: RUST_INPUTS.into_vecs(),
        // Path of output generated Dart code
        dart_output: dart_outputs,
        wasm: true,
        rust_output: Some(rust_outputs),
        // dart_decl_output: Some("../lib/bridge_definitions.dart".into()),   /// would never gen .web
        dart_format_line_length: 120,
        class_name: Some(CLASS_NAMES.into_vecs()),
        // for other options use defaults
        ..Default::default()
    };
```
* some code change
https://github.com/huang12zheng/flutter_rust_bridge/commit/a12ddc9e2bc5f4aa6c4c52ebeba30f21cb27415c

* expect
would generator .web file
> ps My final scenario is in a multi-file case

### PR DESCRIPTION
**Pull request by huang12zheng**
*Please list issues fixed by this PR here, using format "Fixes #the-issue-number".*

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [x] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

## Remark for PR creator
fix #860 

# Cause of some code
I don't really know what I found, you know -_-||
```
error: boolean to int conversion using if
  --> frb_rust/src/support.rs:92:9
   |
92 |         if data { 1_u8 } else { 0_u8 }.into()
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with from: `u8::from(data)`
   |
```
```
error: use of `.unwrap_or_else(..)` to construct default value
   --> frb_codegen/src/config.rs:213:30
    |
213 |       let llvm_compiler_opts = raw
    |  ______________________________^
214 | |         .llvm_compiler_opts
215 | |         .clone()
216 | |         .unwrap_or_else(|| "".to_string());
    | |__________________________________________^
```
```
error: the borrowed expression implements the required traits
  --> frb_codegen/src/source_graph.rs:36:27
   |
36 |         cmd.manifest_path(&manifest_path);
   |                           ^^^^^^^^^^^^^^ help: change this to: `manifest_path`
   |
   = note: `-D clippy::needless-borrow` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```

### CODE REVIEW

File: book/src/feature/wasm.md [hunk 1]
```diff
@@ -30,3 +30,8 @@ flag as well.
 
 Check out [Integrating with Web](../integrate/web.md) for instructions
 on how to consume the web bridge.
+
+What's more, you don't have to use --dart-decl-output,
+you can just >use --wasm to work.
```
Comment (fzyzcjy) [inline book/src/feature/wasm.md:35]:
```suggestion
you can just use --wasm to work.
```

Comment (fzyzcjy) [reply book/src/feature/wasm.md:35]: (only suggest change if it is a typo)

**Review by fzyzcjy — COMMENTED** [review]
uber-niots

**Comment by fzyzcjy** [comment]
@Desdaemon I see you upvoted but not `approve the PR` - so for double check, is it LGTM for you? If so I will merge

File: book/src/feature/wasm.md [hunk 2]
```diff
@@ -30,3 +30,8 @@ flag as well.
 
 Check out [Integrating with Web](../integrate/web.md) for instructions
 on how to consume the web bridge.
+
+What's more, you don't have to use --dart-decl-output,
```
Comment (Desdaemon) [inline book/src/feature/wasm.md:34]: There's a line above that shows the `wasm` flag together with `dart-decl-output`, could you please adjust it to make it consistent? Then this line shouldn't be needed.

**Review by Desdaemon — COMMENTED** [review]
LGTM, before merging please check if omitting `dart-decl-output` for other examples also works. (No need to modify anything for that!)

File: book/src/feature/wasm.md (hunk 2 above)
Comment (huang12zheng) [reply book/src/feature/wasm.md:34]: I can't understand :/ Can you give me an example?
... -_-||

Comment (huang12zheng) [reply book/src/feature/wasm.md:34]: Or just modify it

**Comment by huang12zheng** [comment]
@fzyzcjy I see that you have given a feedback, so can you tell me the meaning of that dialogue?

**Comment by fzyzcjy** [comment]
The wasm is added by @Desdaemon so you may ask him for more details (as I do not remember the details exactly now).

**Comment by huang12zheng** [comment]
@fzyzcjy  @Desdaemon  Check whether the modification([9b00aef](https://github.com/fzyzcjy/flutter_rust_bridge/pull/861/commits/9b00aef2d86da0af9ad79ba9d5096564efc22340)) meets requirements

**Comment by huang12zheng** [comment]
I saw his(Desdaemon's) feedback, which was great for me.
I apologize for the delay in understanding the meaning of the conversation. ;-)

**Comment by fzyzcjy** [comment]
(I will review soon and please ping me if I forget)

**Comment by huang12zheng** [comment]
@fzyzcjy give you a ping

**Comment by fzyzcjy** [comment]
@all-contributors please add @huang12zheng for code, doc

**Pull request merged**